### PR TITLE
Invoices has to be paid to be viewed

### PR DESCRIPTION
**Pull request by mantanquzzmo**
[PT](https://www.pivotaltracker.com/story/show/170400244)

- Adds restriction to show page for invoices

### CODE REVIEW

File: app/policies/application_policy.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  def edit?
+    update?
+  end
+
+  def destroy?
+    false
+  end
+
+  class Scope
```
Comment (oliverochman) [inline app/policies/application_policy.rb]: why do you need this second class?

File: app/policies/application_policy.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+class ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
```
Comment (oliverochman) [inline app/policies/application_policy.rb:4]: do you need all of these policies, stay in scope

File: features/invoices_has_to_be_paid.feature [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  As a company
+  In order to get paid 
+  I must restrict access to invoices
+
+  Background:
+    Given the following user exist
+      | email         | password |
+      | user@user.com | password |
+
+    And I am logged in as "user@user.com"
```
Comment (oliverochman) [inline features/invoices_has_to_be_paid.feature]: this should be underneath the `Given the following invoices exist`

File: features/invoices_has_to_be_paid.feature [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    Given the following invoices exist
```
Comment (oliverochman) [inline features/invoices_has_to_be_paid.feature]: And

File: features/step_definitions/background_steps.rb [hunk 5]
```diff
@@ -27,6 +27,12 @@
   end
 end
 
+Given("the following invoices exist") do |table|
+  table.hashes.each do |invoice_attrs|
+    create(:invoice, invoice_attrs.merge(user: @user))
```
Comment (oliverochman) [inline features/step_definitions/background_steps.rb]: `user = User.find_by(email: company_attrs[:user]) || create(:user, email: company_attrs[:user])`
please use a solution like this for the user

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/policies/application_policy.rb (hunk 1 above)
Comment (mantanquzzmo) [reply app/policies/application_policy.rb]: resolved

File: app/policies/application_policy.rb (hunk 2 above)
Comment (mantanquzzmo) [reply app/policies/application_policy.rb:4]: resolved

File: features/invoices_has_to_be_paid.feature (hunk 3 above)
Comment (mantanquzzmo) [reply features/invoices_has_to_be_paid.feature]: Test need that order to create invoices

File: features/invoices_has_to_be_paid.feature (hunk 4 above)
Comment (mantanquzzmo) [reply features/invoices_has_to_be_paid.feature]: ok

File: features/step_definitions/background_steps.rb (hunk 5 above)
Comment (mantanquzzmo) [reply features/step_definitions/background_steps.rb]: ok

File: features/invoices_has_to_be_paid.feature (hunk 3 above)
Comment (mantanquzzmo) [reply features/invoices_has_to_be_paid.feature]: not anymore

**Review by oliverochman — APPROVED** [review]

**Pull request merged**